oggm/utils.py

```diff
@@ -79,32 +79,40 @@
 # Function
 tuple2int = partial(np.array, dtype=np.int64)
 
+# Global Lock
+lock = mp.Lock()
+
 
 def _get_download_lock():
-    try:
-        lock_dir = cfg.PATHS['working_dir']
-    except:
-        lock_dir = cfg.CACHE_DIR
-    mkdir(lock_dir)
-    lockfile = os.path.join(lock_dir, 'oggm_data_download.lock')
-    try:
-        return filelock.FileLock(lockfile).acquire()
-    except:
-        return filelock.SoftFileLock(lockfile).acquire()
+    return lock
 
 
 def _urlretrieve(url, ofile, *args, **kwargs):
+    p = None
     try:
-        return urlretrieve(url, ofile, *args, **kwargs)
+        cache_dir = cfg.PATHS['dl_cache_dir']
+        if cache_dir and os.path.isdir(cache_dir):
+            p = urlparse(url)
+            p = os.path.join(cache_dir, p.netloc, p.path[1:])
+            if not os.path.exists(os.path.dirname(p)):
+                os.makedirs(os.path.dirname(p))
+            # TODO: Maybe figure out a way to verify the integrity of the cached file?
```

Comment:
Yes, this would be awesome! In particular, there are three possible scenarios:
- the file is downloaded properly
- the file download is aborted and the the file is corrupted. How to check for this?
- the file doesn't exist online (this is a legitimate request from some coastal glaciers that implie trying to download a file without knowing that it doesn't exist. Currently, the failing download is accepted as legit)